File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int j = 0;
        int slen = s.length();
        for(int i = 0; i < t.length(); i++){
            if(j >= slen) return true;
            if(t.at(i) == s.at(j)) j++;
        }
        return j == slen;
    }
};

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        slength = len(s)
        sindex = 0
        for i in range(len(t)):
            if sindex == slength:
                return True
            elif s[sindex] == t[i]:
                sindex += 1
        
        return sindex == slength